### ISSUE TITLE
Caching troubles & questions

### ISSUE THREAD

**Issue by MGough** [conda-incubator/setup-miniconda#65]
I'm trying to take advantage of caching of dependencies. It seems like whatever I do the`setup-miniconda` action takes just as long regardless of whether I've cached my dependencies or not.

```yaml
    - name: Cache conda
      uses: actions/cache@v1
      env:
        # Increase this value to reset cache if environment.yml has not changed
        CACHE_NUMBER: 5
      with:
        path: ~/conda_pkgs_dir
        key: ${{ runner.os }}-conda-${{ env.CACHE_NUMBER }}-${{ hashFiles('environment.yml') }}
    - uses: goanpeca/setup-miniconda@v1
      with:
        activate-environment: myenvname
        channel-priority: strict
        environment-file: environment.yml
        use-only-tar-bz2: true # IMPORTANT: This needs to be set for caching to work properly!
```
Have I made a mistake here that is preventing `setup-miniconda` from loading the packages from `~/conda_pkgs_dir` as specified in the docs? 

**Important note** Possibly worth documenting, removing `python-version: 3.7` from the step resulted in a faster build, but still no advantage when caching. I was seeing much slower build times as the step would hang on installing python 3.7.

My environment file looks a bit like this:
```yaml
name: myenvname
channels:
  - pytorch
  - anaconda
  - conda-forge
  - defaults
dependencies:
  - python=3.7
  - jupyter=1.0.0
  - numpy=1.17.4
  - pandas=0.25.3
  - pytorch=1.4.0
  - pip=20.0.2
  - pip:
    - coverage==5.0.3
```

It seems like some dependencies are downloading regardless of the fact that they were stored in the cache (the caching step is successful and it shows a reasonable size).
```zsh
# Updating conda environment from yaml file...
##############################################

/usr/share/miniconda/condabin/conda env update --file environment.yml --name myenvname
Collecting package metadata (repodata.json): ...working... done
Solving environment: ...working... done
Preparing transaction: ...working... done
Verifying transaction: ...working... done
Executing transaction: ...working... done
Ran pip subprocess with arguments:
['/usr/share/miniconda/envs/my-org/bin/python', '-m', 'pip', 'install', '-U', '-r', '/home/runner/work/my-org/my-repo/condaenv.p1divyve.requirements.txt']
Pip subprocess output:
Collecting coverage==5.0.3
  Downloading coverage-5.0.3-cp37-cp37m-manylinux1_x86_64.whl (227 kB)
```

It seems like it's not picking up the files that were restored from the cache, what am I missing?
Also should the updated `cache@v2` be supported?

**Comment by MGough** [comment]
Oh, I missed a comment in my earlier search. It wasn't clear to me that the pip packages are separately cached even when using conda. I'll try also caching those

https://github.com/goanpeca/setup-miniconda/issues/54#issuecomment-655187159

**Comment by goanpeca** [comment]
Hi @MGough, thanks for the report:

>It wasn't clear to me that the pip packages are separately cached even when using conda.

Yes, that is correct.

Looking at your env file it seems you are mixing quite a bunch of channels, so:
```
  - pytorch
  - anaconda
  - conda-forge
  - defaults
```

First: `anconda` and `defaults` are the same, except `defaults` is served statically (and faster) whereas `anaconda` is mirrored in to anaconda.org/anaconda, serve dynamically (and hence slower)

I would suggest using just defaults.

Second: Why are you using defaults and conda-forge? Normally you would use one or the other since there are binary incompatibilities on some packages. 

How about something like:

```yaml
name: myenvname
channels:
  - pytorch
  - conda-forge
dependencies:
  - python=3.7
  - jupyter=1.0.0
  - numpy=1.17.4
  - pandas=0.25.3
  - pytorch=1.4.0
  - coverage==5.0.3
```

?

**Comment by MGough** [comment]
Thanks for the quick response @goanpeca - Shortly after sending I realised that the problems I was seeing weren't related directly to your action, rather they are design decisions by Conda, but it may be worth mentioning some of them in the documentation to avoid having to deal with lost GitHub users like me 😆.

As for the different channels, I do not know why they are all there. I appreciate the insight into the differences between them and I'll certainly try to see about cutting them back. I think I'll also need to consider breaking my actual `environment.yml` into a few different files, as currently the one I'm using is covering too many cases.

**Comment by goanpeca** [comment]
No problems, please share any ideas on how to improve the README to make things clearer!

**Comment by MGough** [comment]
Thanks again for the suggestion, it helped send me in the right direction for tidying up the file. I can hopefully make some small README suggestions in the next few days.

For anyone else that stumbles upon this:
Currently I've found that the caching doesn't save much time, likely due to the solving of a large number of packages (there are more than in the example I provided) taking a long period of time. I plan on reducing the number of packages, a lot of the python ML packages have some questionable dependency decisions, but for now I found that running `mamba` with `strict` channel priority and no caching is the fastest option.

**Comment by MGough** [comment]
Sorry for the delay in getting back to you. I considered opening a small PR but I think you might be able to find a wording that you're happy with more easily if I just make a suggestion here.

I would suggest a very small addition to the end of the 'Caching' section of the README, which just makes it clear that pip dependencies are cached separately from conda, I know it's not due to how this action works but it may save some confusion for those that aren't used to using conda, or aren't used to how it caches packages.

I'd suggest something like this:
> If you are using pip to resolve any dependencies in your conda environment then you may want to [cache those dependencies separately](https://docs.github.com/en/actions/language-and-framework-guides/using-python-with-github-actions#caching-dependencies), as they are not included in the conda package cache.

My other thought would be to remove `python-version` from the `caching-example` and instead move it within `etc/example-environment.yml`. When I did this I saw a speed increase, although perhaps you have a reason for not doing this?
